### ISSUE TITLE
dr/cpp faultreceiver output

### ISSUE THREAD

**Issue by sebwolf-de** [SeisSol/SeisSol#500]
**Describe the bug**
Fault receivers are not found, if they lie on the wrong side of the fault. The output later crashes

**Expected behavior**
Fault Receiver Output works, no matter on which side of the fault the receiver is positioned.

**To Reproduce**
1. dr/cpp
2. N/A
3. N/A
4. https://github.com/7bits-register/precomputed-seissol/tree/master/tpv5: Replace y component in FaultReceivers.dat with +/- 1e-10

**Additional context**
[This function](https://github.com/SeisSol/SeisSol/blob/dr/cpp/src/DynamicRupture/Output/Builders/PickPointBuilder.hpp#L117-L120) only returns (volume) elements on one side of the fault. Later, for each faultreceiver, we search among this list in which (volume) element the receiver lies. Possible solution: this function returns all (volume) elements on both sides of the fault.

**Comment by ravil-mobile** [comment]
is it the same for Fortran?

**Comment by ravil-mobile** [comment]
```
std::tuple<std::vector<Vertex>,
             std::vector<Element>,
             std ::unordered_map<size_t, std::vector<size_t>>>
      getElementsAlongFault() {
    const auto& fault = meshReader->getFault();
    const auto numFaultElements = fault.size();

    auto meshElements = meshReader->getElements();
    auto meshVertices = meshReader->getVertices();

    std::vector<Vertex> faultVertices(4 * numFaultElements);
    std::vector<Element> faultElements(numFaultElements);

    // note: an element can have multiple fault faces
    std::unordered_map<size_t, std::vector<size_t>> elementToFault{};

    for (size_t faultIdx{0}; faultIdx < numFaultElements; ++faultIdx) {
      const auto& faultItem = fault[faultIdx];

      // element copy done on purpose because we are recording
      // a new vertex array and thus we need to modify vertex indices
      // inside of each element
      std::array<Element*, 2> elements{&meshElements[faultItem.element],
                                       &meshElements[faultItem.neighborElement]};
      for (auto* element: elements) {
        for (size_t vertexIdx{0}; vertexIdx < 4; ++vertexIdx) {
          const size_t faultVertexIdx = vertexIdx + 4 * faultIdx;
          faultVertices[faultVertexIdx] = meshVertices[element->vertices[vertexIdx]];
          element->vertices[vertexIdx] = faultVertexIdx;
        }
        faultElements[faultIdx] = *element;
      }
      elementToFault[faultItem.element].push_back(faultIdx);
    }

    return std::make_tuple(faultVertices, faultElements, elementToFault);
  }
```

what do you think?

**Comment by ravil-mobile** [comment]
I found out that `+1e-10` results in a case when a receiver is out of the domain.

**Comment by sebwolf-de** [comment]
But this can not be the case. I suppose this is because, we don't consider the elements on both sides of the fault.

**Comment by sebwolf-de** [comment]
This has been fixed by #505, right?
